staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go

```diff
@@ -71,12 +71,13 @@ func (f AttrFunc) WithFieldMutation(fieldMutator FieldMutationFunc) AttrFunc {
 
 // SelectionPredicate is used to represent the way to select objects from api storage.
 type SelectionPredicate struct {
-	Label       labels.Selector
-	Field       fields.Selector
-	GetAttrs    AttrFunc
-	IndexFields []string
-	Limit       int64
-	Continue    string
+	Label               labels.Selector
+	Field               fields.Selector
+	GetAttrs            AttrFunc
+	IndexFields         []string
+	Limit               int64
+	Continue            string
+	AllowWatchBookmarks bool
```

Comment:
Can we move adding this field and propagating it down to cacheWatcher to a separate PR?
This one is something we can merge independently and this is a change in the interface so I would prefer to have it separately for better visibility.

Reply:
Moving this to a separate PR seems inappropriate as we do not merge the API code.